packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -26,41 +27,112 @@ class Plugin {
   Plugin({
     this.name,
     this.path,
-    this.androidPackage,
-    this.iosPrefix,
-    this.macosPrefix,
-    this.pluginClass,
+    this.platformPlugins,
   });
 
+  /// Parses [Plugin] specification from the provided pluginYaml.
+  ///
+  /// This currently supports two formats. V1 (Legacy) and V2 (Multi-platform).
+  /// Example V1 format. Which is deprecated.
+  /// flutter:
+  ///  plugin:
+  ///    androidPackage: io.flutter.plugins.sample
+  ///    iosPrefix: FLT
+  ///    macosPrefix: FLT
```

Comment:
Remove; we should not consider macOS to be part of the legacy format. There are ~no public macOS plugins that I don't control (and those that I do have deliberately not been published formally), and I've always been clear that things on desktop are still subject to breakage without fallback. I would prefer that when this lands, only the new format supports macOS, for consistency with the rest of desktop and to minimize legacy code.

(If you can coordinate with me when you land this so I can fix FDE plugins at the same time that would be helpful, but it's not critical.)